File: 3.ConvertDecimaltoBinary/JAVA/src/Main.java
import java.io.*;
import com.google.gson.Gson;
import com.google.gson.stream.JsonReader;

class Config {
    private String testcasesFile;
    private String JavaResultFile;

    String getTestCaseFile() {
        return this.testcasesFile;
    }

    String getOutputFile() {
        return this.JavaResultFile;
    }

    public void setTestCaseFile(String testcasesFile) {
        this.testcasesFile = testcasesFile;
    }

    public void setJavaResultFile(String javaResultFile) {
        JavaResultFile = javaResultFile;
    }
}

class Testcase {
    private int input;
    private String output;
    private boolean result;

    public int getInput() {
        return input;
    }

    public String getOutput() {
        return output;
    }

    public boolean getResult() {
        return result;
    }

    public void setResult(boolean result) {
        this.result = result;
    }

    public void setInput(int input) {
        this.input = input;
    }

    public void setOutput(String output) {
        this.output = output;
    }

    public String toString() {
        return "Input: " + input + " Output: " + output;
    }
}

class Util {
    Testcase[] testcases;
    Config config;
    Gson gson;

    Util() {
        this.gson = new Gson();
        this.config = readConfig();
        this.testcases = readTestcases();
    }

    private Config readConfig() {
        Config config = null;
        try {
            File file = new File("../../conf/config.json");
            JsonReader r = new JsonReader(new FileReader(file.getAbsolutePath()));
            config = gson.fromJson(r, Config.class);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return config;
    }

    private Testcase[] readTestcases() {
        Testcase[] testcases = null;
        try {
            File file = new File("../../" + config.getTestCaseFile());
            JsonReader reader = new JsonReader(new FileReader(file.getAbsolutePath()));
            testcases = gson.fromJson(reader, Testcase[].class);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return testcases;
    }

    public void writeTestCases() {
        try {
            File file = new File("../../" + config.getOutputFile());
            String json = gson.toJson(testcases);
            FileWriter writer = new FileWriter(file.getAbsolutePath());
            writer.write(json);
            writer.close();
            return;
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

public class Main {
    public static void main(String[] args) throws Exception {
        try {
            Util util = new Util();
            DecimalToBinaryTest tester = new DecimalToBinaryTest();
            boolean failed = false;
            for (Testcase testcase : util.testcases) {
                boolean res = tester.test(testcase.getInput(), testcase.getOutput());
                failed = !res;
                testcase.setResult(res);
            }
            util.writeTestCases();
            if (failed)
                throw new RuntimeException("Testcases Failed");
            System.out.println("All Testcases Passed");
        } catch (Exception e) {
            System.out.println(e);
            System.exit(1);
        }
    }
}

File: 3.ConvertDecimaltoBinary/PYTHON/main.py
from DecimalToBinaryTest import DecimalToBinaryTest
import json
import os

class Util:
    @staticmethod
    def read_json(file_path):
        with open(file_path, 'r') as file:
            return json.load(file)

    @staticmethod
    def write_json(file_path, data):
        with open(file_path, 'w') as file:
            json.dump(data, file, indent=4)

class DecimalToBinaryTest:
    def __init__(self):
        self.util = Util()
        self.config = self.util.read_json('conf/config.json')
        self.testcases = self.util.read_json(self.config.get('testcasesFile'))

    def test(self, input, expected_output):
        # Your test logic here
        pass

def main():
    tester = DecimalToBinaryTest()
    testcasesJson = tester.testcases

    for testcase in testcasesJson:
        res = tester.test(testcase['input'], testcase['output'])
        testcase["result"] = "pass" if not res else "fail"

    # Update the result in PythonResultFile
    tester.util.write_json(tester.config.get('PythonResultFile'), testcasesJson)

    if "fail" in [testcase["result"] for testcase in testcasesJson]:
        raise Exception("Testcases failed")
    print("All testcases passed")

if __name__ == "__main__":
    main()
